src/utils/async-event-emitter.ts

```diff
@@ -2,23 +2,24 @@ import Emittery from 'emittery';
 
 export default class AsyncEventEmitter extends Emittery {
     public once (event: string, listener?: Function): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
-        return new Promise((resolve, reject) => {
-            const off = this.on(event, data => {
-                try {
-                    off();
+        if (!listener)
+            return super.once(event);
 
-                    const result = listener ? listener.call(this, data) : data;
+        const unsubscribe = this.on(event, async data => {
+            unsubscribe();
 
-                    resolve(result);
+            return listener(data);
+        });
 
-                    return result;
-                }
-                catch (e) {
-                    reject(e);
+        return Promise.resolve();
+    }
 
-                    throw e;
-                }
-            });
-        });
+    public emit (eventName: string, ...args: unknown[]): Promise<void> {
+        const emitPromise = super.emit(eventName, ...args);
+
+        if (eventName !== 'error')
+            emitPromise.catch(reason => this.emit('error', reason));
```

Comment:
I started to think that we make here the same error duplication mistake that was present in the `once` method before. We should choose a single path: only emit an error event or only allow to propagate via `catch` handlers, not both.

Reply:
We decided to add the `captureRejections` flag to the class constructor that should enable this functionality.